app/javascript/avatar_generator/AvatarGenerator.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect } from 'react'
+const Axios = require('axios');
+import AvatarDrawing from './AvatarDrawing.jsx';
+import AvatarMenu from './AvatarMenu.jsx';
+
+function AvatarGenerator(props) {
+  const [hat, setHat] = useState({
+    hatType: "none",
+    hatSize: 5,
+    hatColor: "#666666",
+  });
+  const [face, setFace] = useState({
+    faceType: "smile",
+    faceSize: 5,
+    faceColor: "#666666",
+  });
+  const [shirt, setShirt] = useState({
+    shirtOn: false,
+    shirtColor: "#666666",
+  });
+
+  function getUserId() {
+    let userId = window.location.pathname.split("/")[2];
```

Comment:
Tagad redzu kā tas nebija laba doma ņemt no url. Tāpēc kā var radities daudz problemu...

Ir divi varianti kā tu to vari atrisināt. Pirmais: kaut kur lapas data attributos, kur tev tiek ieladets React, saglabat user_id

Otrs, sadiskaks, paņemt ar Javascript X-Auth-header (vai kaut kā tā) un kad prasisi avataru, padot headeri ar axious, lai to atpazistu

Reply:
Nevareju atrast nekadus rakstus saistitus ar devise un tokeniem, tikai kkads jwt kuru apvienojot ar devise tiek razoti tokeni, ta vismaz sapratu. Links uz to repo : https://github.com/waiting-for-dev/devise-jwt

Reply:
Viss ok. Atradu, tas bija iekš session objektā. Un JSON requesti normali visu saprot.

Ieliec šo, kur to vajag

render json: current_user.avatar_bundle